src/vss_tools/vspec/vssexporters/vss2samm/vss2samm.py

```diff
@@ -0,0 +1,291 @@
+def __get_version_vss(vss_tree: VSSNode):
+    # DEFAULT version would be 1.0.0
+    major = 1
+    minor = 0
+    patch = 0
+
+    if vss_tree and hasattr(vss_tree, "children") and len(vss_tree.children) > 0:
+        # Get the VersionVSS node so to extract current VSS version from it.
+        vss_version_node = next(filter(lambda vs_child: vs_child.name == 'VersionVSS',  vss_tree.children), None)
+
+        if vss_version_node:
+            for v_child in vss_version_node.children:
+                if v_child.name in ['Major', 'Minor', 'Patch'] \
+                    and hasattr(v_child, 'default') \
+                        and type(v_child.default) is int and v_child.default > 1:
+
+                    match v_child.name:
+                        case 'Major':
+                            major = v_child.default
+                        case 'Minor':
+                            minor = v_child.default
+                        case 'Patch':
+                            patch = v_child.default
+
+    return f"{major}.{minor}.{patch}"
+
+
+# Exporter specific options
+
+@clo.option(
+    '-sigf', '--signals-file', 'signals_file', show_default=False,
+    help="""\b
+Path to file with selected VSS signals to be converted.
+Allows to convert just selected VSS signals into aspect model(s), if '-spl / --split' is enabled.
+\033[36mNOTE:\033[0m each signal in the file should be on a new line and in the format of:
+      \033[96mPARENT_SIGNAL.PATH.TO.CHILD_SIGNAL\033[0m as defined in VSS.
+\033[33mEXAMPLE:\033[0m \033[96m-sigf PATH_TO_FILE/selected_signals.txt\033[0m
+ """
+)
+@clo.option(
+    "--split-depth", "-spld", "split_depth",
+    required=False, default=1, show_default=False,
+    help="""\b
+Number - used to define, up to which level, VSS branches will be converted into single aspect models.
+Can be used in addition to the \033[32m-spl, --split\033[0m option.
+Default value of 1 means that only 1st level VSS branches like Vehicle.Cabin, Vehicle.Chassis etc.,
+will be converted to separate aspect models.
+\033[30m[default: 1]\033[0m
+ """
+)
+@clo.option(
+    "--split", "-spl", "split",
+    required=False, is_flag=True, default=False, show_default=False,
+    help="""\b
+Boolean flag - used to indicate whether to convert VSS specifications in separate ESMF Aspect(s)
+or the whole (selected) VSS specification(s) will be combined into single ESMF Aspect model.
+\033[30m[default: False]\033[0m
+ """
+)
+@clo.option(
+    "--target-namespace", "-tns", "output_namespace",
+    required=False,
+    default='com.covesa.vss.spec',
+    show_default=False,
+    help="""\b
+Namespace for VSS library, located in specified '--target-folder'.
+Will be used as name of the folder where VSS Aspect models (TTLs) are to be stored.
+This folder will be created as subfolder of the specified '--target-folder' parameter.
+\033[30m[default: com.covesa.vss.spec]\033[0m
+ """
+)
+# TODO: shall we use Path, similar to output option, or just a simple string here?
```

Comment:
Just a question here.

With this setup for target folder:

type=click.Path(dir_okay=True, writable=True, path_type=Path),

would it be possible to pass just a string, and then the script will create this folder dynamically or user will have to have created the target folder in advance?

Because with current implementation and using os I am able to create dynamically the specified target folder and with respect to location from which the script was called.

Reply:
Done.